### PR TITLE
fix #67

### LINKED ISSUE #67: Uncaught TypeError: Path must be a string. Received undefined
1. Install via terminal
2. Launch Editorconfig via Command Palette
3. It fails with below info

**Atom Version**: 1.5.4
**System**: Mac OS X 10.11.3
**Thrown From**: [editorconfig](https://github.com/sindresorhus/atom-editorconfig) package, v1.2.4

### Stack Trace

Uncaught TypeError: Path must be a string. Received undefined

```
At path.js:8

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:8:11)
    at Object.posix.join (path.js:477:5)
    at atom-workspace.init (/Users/Kruse/.atom/packages/editorconfig/commands/generate.js:6:26)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at CommandPaletteView.module.exports.CommandPaletteView.confirmed (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/command-palette/lib/command-palette-view.js:183:32)
    at CommandPaletteView.module.exports.SelectListView.confirmSelection (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:338:21)
    at space-pen-div.atom.commands.add.core:confirm (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:109:19)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:570:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:378:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -1:00.5.0 settings-view:open (atom-text-editor.editor.is-focused)
     -0:04.5.0 command-palette:toggle (div.package-detail.panels-item)
     -0:00.8.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:00.8.0 EditorConfig:generate-config (div.package-detail.panels-item)
```

### Config

```json
{
  "core": {}
}
```

### Installed Packages

```coffee
# User
color-picker, v2.1.1
editorconfig, v1.2.4
file-icons, v1.6.18
floobits, v0.31.0
git-log, v0.4.1
highlight-line, v0.11.1
minimap, v4.20.0
minimap-git-diff, v4.2.0

# Dev
No dev packages
```

### PR DESCRIPTION
**Pull request by florianb**
I had to export `generate.init()` to test the behavior. I tried first to `rewire` the module but rewire doesn't seem to work with babel.

I couldn't find out if there's another way to access init, without exposing it in general.

@sindresorhus i'll immediately merge that in and release the overall-patch if you give the go: 👍

### CODE REVIEW

File: commands/generate.js [hunk 1]
```diff
@@ -3,7 +3,17 @@ import fs from 'fs';
 import path from 'path';
 
 const init = () => {
-	const configFile = path.join(atom.project.getPaths()[0], '.editorconfig');
+	let basePath = '';
+	if (atom.project.getPaths().length > 0) {
```
Comment (Arcanemagus) [inline commands/generate.js]: Call once and store the results, no need to spend the time calling it twice.

Comment (florianb) [inline commands/generate.js]: About which call do you talk?

File: index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 const lazyReq = require('lazy-req')(require);
 
 const editorconfig = lazyReq('editorconfig');
-const generateConfig = lazyReq('./commands/generate');
+import generateConfig from './commands/generate';
```
Comment (sindresorhus) [inline index.js]: Even though Babel doesn't care, `import` should always be at the top.

**Comment by sindresorhus** [comment]
@florianb Looks good ;)

In the future would you mind giving PRs a proper title? Looking at `fix #37` in the commit history isn't very helpful ;)

**Comment by florianb** [comment]
@sindresorhus thanks - yeah, you're right i need to spend more care at that point. I usually rename it when squashing - i just had too much in my mind. I will try to amend it.

After that i'll publish 1.2.5. 👍

**Comment by sindresorhus** [comment]
1.3.0 published.

Thanks for all the time you've spent on triaging and fixing issues @florianb. I really appreciate it :)

**Comment by florianb** [comment]
@sindresorhus you're welcome. I'll try to work towards solving the last big issue in the next week.

File: commands/generate.js (hunk 1 above)
Comment (Arcanemagus) [inline commands/generate.js]: `atom.project.getPaths()`, you can call that once, store the results, then do the check on the `.length` here and the `[0]` element on the next line without re-calling the function.

Comment (florianb) [inline commands/generate.js]: Yeah thats true - i'll keep that in mind for the next refactoring.

**Pull request merged**